src/main/java/com/arpnetworking/metrics/mad/parsers/ProtobufV3ToRecordParser.java

```diff
@@ -70,6 +70,7 @@
                         DefaultRecord.Builder.class,
                         builder -> builder.setId(new UUID(high, low).toString())
                                 .setTime(ZonedDateTime.ofInstant(Instant.ofEpochMilli(record.getEndMillisSinceEpoch()), ZoneOffset.UTC))
+                                .setRequestTime(ZonedDateTime.ofInstant(Instant.ofEpochMilli(record.getRequestMillisSinceEpoch()), ZoneOffset.UTC))
```

Comment:
This probably should not be called if the record does not have a `requestMillisSinceEpoch`.

Reply:
Seems likethe way this code is organized means that I'd have to copypaste the whole Builder block if I wanted to add an if statement. What do you think of this?

```
.setRequestTime(maybeParseMillisSinceEpoch(record.getRequestMillisSinceEpoch()).orElse(null))
```

Reply:
No, that should not be necessary, just convert the single line lambda into a method; e.g.

```
records.add(ThreadLocalBuilder.build(
    DefaultRecord.Builder.class,
    builder -> {
           builder.setId(new UUID(high, low).toString())
               .setTime(ZonedDateTime.ofInstant(
                   Instant.ofEpochMilli(record.getEndMillisSinceEpoch()), ZoneOffset.UTC))
               .setDimensions(buildDimensions(record))
               .setMetrics(buildMetrics(record))
           if (record.getRequestMillisSinceEpoch() != ) {
               builder.setRequestTime(...);
           }
           return builder;
    }));
```

The above would negate the need for the `maybeParseMillisSinceEpoch` method entirely, since the conditional allows you to encapsulate that code too and that logic is not currently reused. 

However, your method works too!

Reply:
General note; in Java there are three kinds of lambdas:

1) Method reference; e.g. MyClass::doFoo
2) Single expression; e.g. builder -> builder.setFoo(...)
3) Inline method; e.g. builder -> { builder.setFoo(...); if (x) { builder.setBar(...) }; return builder }

Oracle reference here:

https://docs.oracle.com/javase/tutorial/java/javaOO/lambdaexpressions.html#syntax

(not a terrific read imo; better off if you're interested in picking up Effective Java v3).

Reply:
ugh, I wanted to do the *clever* *functional* Optional-using thing, but honestly your suggested lambda is much more concise and easier to read.